pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -110,6 +118,7 @@ public void shutDown() {
   @Override
   public void addSegment(@Nonnull ImmutableSegment immutableSegment) {
     String segmentName = immutableSegment.getSegmentName();
+    untrackIfDeleted(segmentName);
```

Comment:
This line should move down to be after line 128 (when we populate the _segmentDataManagerMap). The segment is available for query only after that point.

Reply:
Done.